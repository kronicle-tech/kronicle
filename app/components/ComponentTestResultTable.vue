<template>
  <table
    v-if="componentsAndTestResults && componentsAndTestResults.length > 0"
    class="table table-dark table-bordered table-striped mt-2"
    style="width: 100%"
  >
    <thead>
      <tr>
        <th>Name</th>
        <th v-if="teamsVisible">Teams</th>
        <th>Test Results</th>
      </tr>
    </thead>
    <tbody>
      <tr
        v-for="componentAndTestResults in componentsAndTestResults"
        :key="componentAndTestResults.component.id"
      >
        <td class="component-name table-primary">
          <ComponentName :component="componentAndTestResults.component" />
        </td>
        <td v-if="teamsVisible">
          <ComponentTeams
            :component-teams="componentAndTestResults.component.teams"
          />
        </td>
        <td>
          <TestResults :test-results="componentAndTestResults.testResults" />
        </td>
      </tr>
    </tbody>
  </table>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue'
import ComponentName from '~/components/ComponentName.vue'
import ComponentTeams from '~/components/ComponentTeams.vue'
import TestResults from '~/components/TestResults.vue'
import { ComponentAndTestResults } from '~/types/component-test-results'

export default Vue.extend({
  components: {
    ComponentName,
    ComponentTeams,
    TestResults,
  },
  props: {
    componentsAndTestResults: {
      type: Array as PropType<ComponentAndTestResults[]>,
      default: undefined,
    },
    teamsVisible: {
      type: Boolean,
      default: true,
    },
  },
})
</script>
