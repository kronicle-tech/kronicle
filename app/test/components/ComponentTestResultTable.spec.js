import { mount } from '@vue/test-utils'
import ComponentTestResultTable from '@/components/ComponentTestResultTable.vue'

describe('ComponentTestResultTable', () => {
  let propsData
  let wrapper
  const createWrapper = () => {
    wrapper = mount(ComponentTestResultTable, { propsData })
  }

  beforeEach(() => {
    propsData = {}
  })

  afterEach(() => {
    wrapper.destroy()
    wrapper = null
  })

  describe('when componentsAndTestResults prop is undefined', () => {
    test('renders nothing', () => {
      createWrapper()
      expect(wrapper.html()).toEqual(``)
    })
  })

  describe('when componentsAndTestResults is set to multiple components with multiple test results', () => {
    beforeEach(() => {
      propsData.componentsAndTestResults = [
        {
          component: {
            id: 'test-id-1',
            name: 'Test Name 1',
          },
          testResults: [
            {
              testId: 'test-test-id-1',
              outcome: 'fail',
              priority: 'low',
              message: 'test-message-1',
            },
            {
              testId: 'test-test-id-2',
              outcome: 'pass',
              priority: 'medium',
              message: 'test-message-2',
            },
          ],
        },
        {
          component: {
            id: 'test-id-2',
            name: 'Test Name 2',
          },
          testResults: [
            {
              testId: 'test-test-id-3',
              outcome: 'not-applicable',
              priority: 'high',
              message: 'test-message-4',
            },
            {
              testId: 'test-test-id-4',
              outcome: 'fail',
              priority: 'very-high',
              message: 'test-message-4',
            },
          ],
        },
      ]
    })

    test('renders the components and their test results', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
