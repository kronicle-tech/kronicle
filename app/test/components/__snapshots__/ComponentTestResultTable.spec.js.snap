// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentTestResultTable when componentsAndTestResults is set to multiple components with multiple test results renders the components and their test results 1`] = `
"<table class=\\"table table-dark table-bordered table-striped mt-2\\" style=\\"width: 100%;\\">
  <thead>
    <tr>
      <th>Name</th>
      <th>Teams</th>
      <th>Test Results</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td class=\\"component-name table-primary\\"><a href=\\"/components/test-id-1\\">Test Name 1</a></td>
      <td>
        <!---->
      </td>
      <td>
        <table class=\\"table table-dark\\">
          <thead>
            <tr>
              <th>Test</th>
              <th>Priority</th>
              <th>Outcome</th>
              <th>Message</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td class=\\"test-id\\"><a href=\\"/tests/test-test-id-1\\">test-test-id-1</a></td>
              <td class=\\"test-priority\\">
                low
              </td>
              <td class=\\"test-outcome table-danger\\">
                fail
              </td>
              <td class=\\"test-message\\">
                <div>
                  <p>test-message-1</p>
                </div>
              </td>
            </tr>
            <tr>
              <td class=\\"test-id\\"><a href=\\"/tests/test-test-id-2\\">test-test-id-2</a></td>
              <td class=\\"test-priority\\">
                medium
              </td>
              <td class=\\"test-outcome table-primary\\">
                pass
              </td>
              <td class=\\"test-message\\">
                <div>
                  <p>test-message-2</p>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </td>
    </tr>
    <tr>
      <td class=\\"component-name table-primary\\"><a href=\\"/components/test-id-2\\">Test Name 2</a></td>
      <td>
        <!---->
      </td>
      <td>
        <table class=\\"table table-dark\\">
          <thead>
            <tr>
              <th>Test</th>
              <th>Priority</th>
              <th>Outcome</th>
              <th>Message</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td class=\\"test-id\\"><a href=\\"/tests/test-test-id-4\\">test-test-id-4</a></td>
              <td class=\\"test-priority\\">
                very-high
              </td>
              <td class=\\"test-outcome table-danger\\">
                fail
              </td>
              <td class=\\"test-message\\">
                <div>
                  <p>test-message-4</p>
                </div>
              </td>
            </tr>
            <tr>
              <td class=\\"test-id\\"><a href=\\"/tests/test-test-id-3\\">test-test-id-3</a></td>
              <td class=\\"test-priority\\">
                high
              </td>
              <td class=\\"test-outcome table-light\\">
                not-applicable
              </td>
              <td class=\\"test-message\\">
                <div>
                  <p>test-message-4</p>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </td>
    </tr>
  </tbody>
</table>"
`;
